src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/CachedServerInfo.cs

```diff
@@ -115,6 +116,80 @@ public static void AddOrUpdateIsAzure(string  dataSource, bool isAzure)
             }
         }
 
+        public static void AddOrUpdateIsSqlDw(IDbConnection connection, bool isSqlDw)
+        {
+            Validate.IsNotNull(nameof(connection), connection);
+
+            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(connection.ConnectionString);
+            AddOrUpdateIsSqlDw(builder.DataSource, isSqlDw);
+        }
+
+        public static void AddOrUpdateIsSqlDw(string dataSource, bool isSqlDw)
```

Comment:
I added an enum to categorize each variable and combined all the logic. It is a departure from the way DacFX was doing it, but I do think this design is more sustainable. (especially if we decide to cache more connection information).